dpe/src/x509.rs

```diff
@@ -1056,10 +1056,6 @@ impl CertWriter<'_> {
     fn encode_bytes(&mut self, bytes: &[u8]) -> Result<usize, DpeErrorCode> {
         let size = bytes.len();
 
-        if self.offset >= self.certificate.len() || self.offset + size > self.certificate.len() {
-            return Err(DpeErrorCode::InternalError);
-        }
-
```

Comment:
Hmm I think we should keep this.

Reply:
Yeah, otherwise a panic will be introduced